### PR TITLE
Proof of concept for suggested restructuring.

### PR DESCRIPTION
**Pull request by noniq**
__WARNING:__ This is untested!

It's sole purpose is to demonstrate the restructuring I suggested. If you think it's a good idea to go in this direction, then I'd be glad to take a deeper look and create a proper pull request.

__Update Apr. 11:__ No longer untested :smile:

### CODE REVIEW

File: lib/middleman-simple-thumbnailer/image.rb [hunk 1]
```diff
@@ -24,22 +25,15 @@ def resized_img_path
...
-      image.resize(resize_to)
-      save_cached_thumbnail
-    end
-
     def save!
-      if cached_thumbnail_available?
-        FileUtils.cp cached_resized_img_abs_path, resized_img_abs_path
-      else
-        image.write(resized_img_abs_path)
-      end
```
Comment (kubenstein) [inline lib/middleman-simple-thumbnailer/image.rb]: here we need to add `resize!` as well to have resizing in build enabled

**Comment by kubenstein** [comment]
thank you for your work @noniq, code looks really nice. i tested it manually and one thing needs to be add, i put comment.
also i guess `require 'fileutils'` is not needed any more.
now only tests wait to be fixed and also we need to have test for this nasty bug - check content of resized files after build. Ill try to take care of those.

File: lib/middleman-simple-thumbnailer/image.rb (hunk 1 above)
Comment (noniq) [inline lib/middleman-simple-thumbnailer/image.rb]: Of course, you're right (was there initially, got lost when I rebased on – note to self: even proof of concepts shouldn't be done in hurry …)

**Comment by noniq** [comment]
Thanks! I'll polish it up later today. For the tests I think the most reliable way would be to use MiniMagick to check the actual dimensions of the generated files. I could try to add such tests to this PR, if you approve.

**Comment by noniq** [comment]
I fixed the tests and splitted the PR into 3 commits. Should all be okay now. You can ignore the last commit if you want to keep the “reuse cached thumbnails on build” behaviour.

**Comment by kubenstein** [comment]
thank you for your work @noniq! i prefer not to reuse cache in build so will merge all code as is. i also checked on my local machine tests against ruby 2.3.0dev and all are green. Not sure why travis failed on that version.

**Pull request merged**